### PR TITLE
rewrite the CLI in node

### LINKED ISSUE #1: rewrite in node.js
Rewrite in node for easier maintenance and consistency across the pelias org.

Related to pelias/pelias#61

### PR DESCRIPTION
**Pull request by sevko**
Because of the CLI's increasing complexity, the fact that shell scripts are hell, and for consistency with the rest of the Pelias codebase, I rewrote it in node.

The other primary change is that the CLI no longer uses repositories' `package.json`s' `scripts` objects as the subcommands configuration. Subcommands are instead configured in per-repo JSON files in `subcommands/`, because of the greater flexibility/clarity it provides us.

Resolve #1.

### CODE REVIEW

File: .jshintrc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "node": true,
+  "curly": true,
+  "latedef": true,
+  "quotmark": true,
+  "undef": true,
+  "unused": true,
+  "trailing": true
+}
```
Comment (missinglink) [inline .jshintrc:21]: This is different from the settings we use in other repos?
ref: https://github.com/pelias/api/blob/master/.jshintrc

**Comment by missinglink** [comment]
It looks like `execSync` was introduced in `0.12`, it would be great to find a `0.10` analogue, otherwise adding an `.engines` section to `package.json` will alert the users that their system in incompatible during installation:
ref: https://docs.npmjs.com/files/package.json#engines

``` bash
$ ./pelias.js api test
pelias: Cloning repo.

/var/www/pelias/cli/pelias.js:23
  return childProcess.execSync( shellCmd ).toString();
                      ^
TypeError: Object #<Object> has no method 'execSync'
    at $ (/var/www/pelias/cli/pelias.js:23:23)
    at runRepoSubcommand (/var/www/pelias/cli/pelias.js:176:5)
    at Object.<anonymous> (/var/www/pelias/cli/pelias.js:191:1)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:929:3
```

``` bash
$ node --version
v0.10.36
```

**Comment by missinglink** [comment]
:+1: looks good, I think we should just revolve the `0.10` issue before releasing.

**Comment by sevko** [comment]
Indeed, Node <0.12 unfortunately doesn't have a built-in `execSync()`. After digging around some I found a polyfill package, [runsync](https://www.npmjs.com/package/runsync), that looks promising (playing around with it locally and nothing's broken thus far). Even though it look like we might be officially moving over to 0.12 across the Pelias board, I'd prefer to have things Just Work with at least 0.10. I reckon we should try out `runsync` and, if it happens to give us problems, we'll just make this tool 0.12+.

**Comment by sevko** [comment]
Unfortunately, I've played around with two polyfill solutions and neither fits the bill because of subtle differences/bugs. Unless we feel very strongly about making this support 0.10, I'll go ahead and publish as is. I've currently added an `engines` fields to `package.json` and the tool checks for the existence of `require("child_process").execSync()` at runtime to confirm that the desired version of Node is being used, aborting otherwise.

**Comment by dianashk** [comment]
We could change the code to run asynchronously, right? Just call regular exec and check the outcome in the callback.

**Comment by sevko** [comment]
Yes, but it seems pretty impractical because everything here is intended to run synchronously so the callback hell would be real.

**Comment by dianashk** [comment]
I suppose. What would your implementation have been like if 0.12.0 wasn't out yet? You'd probably find a solution that didn't involve callback hell.  Just saying it's doable.

However, I'm not sure it's worth the effort at this time since it's not our core focus for the next few weeks at least. So I'm ok with possibly pushing this through as is (add package.json limits for node version) and adding an issue to make compatible with 0.10.*

**Comment by sevko** [comment]
It's probably doable, but after I explored this issue a bit `.exec()` turned out to be the best means of executing a process since it provides shell-like string parsing (which something like `.spawn()` doesn't), and I jumped for `.execSync()` to make the code more manageable.

Also, come to think of it, I'm not sure that asynchronous `.exec()` _would_ work because it buffers all of the child process's output, making it infeasible for long-running procs (like the API or any of the importers). You _can_ read the returned `ChildProcess` object's `stdout`/`stderr` streams, but I remember there were other problems, like an inability to keyboard interrupt the process.

I'm in favor of keeping it as is and opening a 0.10.\* ticket.

**Pull request merged**